### PR TITLE
Deny missing docs

### PR DESCRIPTION
**Pull request by tdejager**
Added deny missing docs and code comments in a lot of places!

### CODE REVIEW

File: crates/rattler_installs_packages/src/resolve.rs [hunk 1]
```diff
@@ -1,3 +1,11 @@
+//! This module contains the [`PypiDependencyProvider`] which is used
```
Comment (baszalmstra) [inline crates/rattler_installs_packages/src/resolve.rs:1]: this is out of date now. There is just the resolve function and the feature is called resolvo.

File: crates/rattler_installs_packages/src/artifact.rs [hunk 2]
```diff
@@ -11,6 +11,8 @@ use std::io::Read;
 use std::str::FromStr;
 use zip::ZipArchive;
 
+/// Trait that represents an artifact type in the PyPi ecosystem.
```
Comment (wolfv) [inline crates/rattler_installs_packages/src/artifact.rs:14]: Super nit-picky but I think it's PyPI and PyPA (in terms of capitalization).

File: crates/rattler_installs_packages/src/resolve.rs (hunk 1 above)
Comment (tdejager) [reply crates/rattler_installs_packages/src/resolve.rs]: Done

File: crates/rattler_installs_packages/src/artifact.rs (hunk 2 above)
Comment (tdejager) [reply crates/rattler_installs_packages/src/artifact.rs]: Done

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**